evaldo/builtins.go

```diff
@@ -6934,6 +6936,36 @@ var builtins = map[string]*env.Builtin{
 		},
 	},
 
+	"os\\open": { // todo -- variation is not meant for grouping inside context ... just for function variations ... just temp to test
+		Argsn: 1,
+		Doc:   "",
+		Fn: func(ps *env.ProgramState, arg0 env.Object, arg1 env.Object, arg2 env.Object, arg3 env.Object, arg4 env.Object) env.Object {
+			switch s0 := arg0.(type) {
+			case env.Uri:
+
+				var err error
+				url := s0.GetFullUri(*ps.Idx)
+
+				switch runtime.GOOS {
+				case "linux":
+					err = exec.Command("xdg-open", url).Start()
+				case "windows":
+					err = exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
+				case "darwin":
+					err = exec.Command("open", url).Start()
```

Comment:
Attention: This could lead to command execution trough [various schemes](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml), possibly beinhg handled by vulnerable programs.

Reply:
Thank you for the comment. I was making a bookmarking repl-app demo and wanted to open a bookmark at the end, so I added this. I suspected it's an open door for danger, so I agree this has to be dealt with. What do you propose? What comes to mind:
A) just don't have this functionality in any case
B) add this only under specific build tag "danger_open_command" for specific cases
D) be very strict about the input (if possible), limit to concrete few (whitelist)
E) make specific command to open a browser, not general os\open like I did
F) entirely different approach ... propose

To me  B + C or E seems to make most sense. What do you think?

Reply:
I see no C in the list. I would avoid more build tags. Possibly limiting the functionality to common `http://` & `https://` schemas. Opening a `file://` could be problematic.

Ge team keeps https://pkg.go.dev/cmd/internal/browser#Open _internal_ for a reason 🤷

Reply:
Missed the C :) . Ok, agree about the tags. I agree on just limit to http(s), but then we just better use cmd/internal/browser from Go or is that not meant to be used in general? 

It could be useful to be able to open a generated or downloaded PNG or PDF (for example a graph drawn with goplot), but we can think about that if the concrete need arises.

Reply:
Yes whitelisting is an option, including whitelisting schemas as suggested.

You can't use Go packages with `internal` in the path from other Go modules. Those packages are only accessible within their module.

Reply:
I commented it in for now, noting todos. The reason was that even the naming or location (module) was not yet decided. I need to think what other similar OS dependant or OS level functions belog next to it. With this removed and todo written about what we talked about I'm marking this resolved. Thank you!!